.github/workflows/tests-ipsec-upgrade.yaml

```diff
@@ -0,0 +1,415 @@
+              # BPF IPv6 masquerade is not supported in < v1.14
+              EXTRAS_V13+=" --helm-set=enableIPv6Masquerade=false"
+            fi
+
+            CILIUM_CLI_MODE=helm ./cilium-cli install \
+              --version ${{ env.upgrade_cilium_from_vsn }} \
+              ${{ steps.vars.outputs.cilium_install_defaults }} \
+              \$EXTRAS_V13
+
+            ./cilium-cli status --wait
+            kubectl get pods --all-namespaces -o wide
+            kubectl -n kube-system exec daemonset/cilium -- cilium status
+
+            # Disable no-missed-tail-calls due to https://github.com/cilium/cilium/issues/26739
+            ./cilium-cli connectivity test --include-unsafe-tests --collect-sysdump-on-failure \
+              --flush-ct \
+              --test '!no-missed-tail-calls' \
+              --sysdump-hubble-flows-count=1000000 --sysdump-hubble-flows-timeout=5m \
+              --sysdump-output-filename "cilium-sysdump-${{ matrix.name }}-<ts>" \
+              --junit-file "cilium-junits/${{ env.job_name }} (${{ join(matrix.*, ', ') }}).xml" \
+              --junit-property github_job_step="Run tests upgrade 1 (${{ join(matrix.*, ', ') }})"
+
+            EXTRA_CONNECTIVITY_TEST=""
+            if [ "${{ matrix.test-flow-interrupts }}" == "true" ]; then
+              # Create pods which establish long lived connections. It will be used by
+              # subsequent connectivity tests with --include-conn-disrupt-test to catch any
+              # interruption in such flows.
+              ./cilium-cli connectivity test --include-conn-disrupt-test --conn-disrupt-test-setup
+              EXTRA_CONNECTIVITY_TEST=" --include-conn-disrupt-test"
+            fi
+
+            # Upgrade
+
+            if [ -z "${{ env.upgrade_cilium_to_vsn }}" ]; then
+              CILIUM_CLI_MODE=helm ./cilium-cli upgrade \
+                ${{ steps.vars.outputs.cilium_main_image_settings }} \
+                ${{ steps.vars.outputs.cilium_install_defaults }}
+            else
+              CILIUM_CLI_MODE=helm ./cilium-cli upgrade \
+                --version ${{ env.upgrade_cilium_to_vsn }} \
+                ${{ steps.vars.outputs.cilium_install_defaults }}
+            fi
+
+            ./cilium-cli status --wait
+            kubectl get pods --all-namespaces -o wide
+            kubectl -n kube-system exec daemonset/cilium -- cilium status
+
+            ./cilium-cli connectivity test --include-unsafe-tests --collect-sysdump-on-failure \
+              \$EXTRA_CONNECTIVITY_TEST \
+              --flush-ct \
+              --test '!no-missed-tail-calls' \
+              --sysdump-hubble-flows-count=1000000 --sysdump-hubble-flows-timeout=5m \
+              --sysdump-output-filename "cilium-sysdump-${{ matrix.name }}-<ts>" \
+              --junit-file "cilium-junits/${{ env.job_name }} (${{ join(matrix.*, ', ') }}).xml" \
+              --junit-property github_job_step="Run tests upgrade 2 (${{ join(matrix.*, ', ') }})"
```

Comment:
Is the `--include-conn-disrupt-test` part of this connectivity tests always executed last? If not, we may want to separate it from the rest to ensure it is executed last. The rationale is that connectivity tests themselves could cause XFRM errors and missed tail calls, so we'd want to check for those after running the tests.

Reply:
The upgrade tests don't actually perform any connectivity tests, right? They just check various counters IIUC. So should be fine w.r.t. L7 CT reuse.

Reply:
The upgrade test checks whether the connections are not interrupted too. So, the L7 tests might interrupt them :-/

Reply:
The L7 tests would interrupt existing connections because of the CT reuse bug? I wasn't aware of that :disappointed:

Reply:
I think I just spread FUD about L7 tests. It might be fine, but maybe let's try to change the order after merging this PR. Also, I haven't tested the suggested order yet.

Reply:
:+1: